packages/python/plotly/plotly/express/_core.py

```diff
@@ -1648,109 +1846,212 @@ def process_dataframe_hierarchy(args):
     """
     Build dataframe for sunburst, treemap, or icicle when the path argument is provided.
     """
-    df = args["data_frame"]
+    df: nw.DataFrame = args["data_frame"]
     path = args["path"][::-1]
     _check_dataframe_all_leaves(df[path[::-1]])
-    discrete_color = False
+    discrete_color = not _is_continuous(df, args["color"]) if args["color"] else False
+
+    df = df.lazy()
 
-    new_path = []
-    for col_name in path:
-        new_col_name = col_name + "_path_copy"
-        new_path.append(new_col_name)
-        df[new_col_name] = df[col_name]
+    new_path = [col_name + "_path_copy" for col_name in path]
+    df = df.with_columns(
```

Comment:
Similar to my comment above, I find this code pretty hard to read compared to the code it's replacing. Maybe if I was more familiar with Polars I'd find it easier.

Reply:
This is definitly a common way in polars to assign a new column, but not completely unknown to pandas users.

If you were to use `pd.DataFrame.assign`, but your don't know if the column name has spaces (since provided by the user), you would still use dictionary unpacking:

```py
df.assign({
   "col with space in name": ...
})
```

since you cannot do `df.assign(col with space in name = ...)`

Reply:
How about rewriting this as
```python
    df = df.with_columns(
        nw.col(col_name).alias(new_col_name)
        for new_col_name, col_name in zip(new_path, path)
    )
```
?

Reply:
@FBruzzesi @MarcoGorelli I think this is a big readability improvement, thank you. 👍